src/main.rs

```diff
@@ -121,6 +123,18 @@ impl<'a> NodeSigner for LndNodeSigner<'a> {
     }
 }
 
+// MessengerUtilities implements some utilites required for onion messenging.
+struct MessengerUtilities {}
+
+impl EntropySource for MessengerUtilities {
+    fn get_secure_random_bytes(&self) -> [u8; 32] {
```

Comment:
Just wondering, is this the suggested implementation of `get_secure_random_bytes`? Seems like this means it will only run on linux?

Reply:
It isn't :( 

LDK has their own impl [on the KeysManager](https://github.com/lightningdevkit/rust-lightning/blob/fac5373687a4c7919c8639744dc712d922082cc3/lightning/src/chain/keysinterface.rs#L1255) which uses the node's seed as entropy + ChaCha's it. Felt like overkill to pull in a whole `KeysManger` for our purposes, and some of the elements used in LDK aren't public so we can't just re-impl it. 

@dunxen suggested pulling in `rand` crate instead, WDYT?

Reply:
Updated to [ChaCha20](https://rust-random.github.io/rand/rand_chacha/struct.ChaCha20Rng.html) crate, uses `dev/random` under the hood to seed. Added a TODO for future to extract `EntropySource` in  LDK and use it instead.